AudioPlayer/AudioPlayer/player/AudioPlayerState.swift

```diff
@@ -25,13 +25,13 @@ public enum AudioPlayerError: Error {
 /// - stopped: The player is stopped.
 /// - waitingForConnection: The player is waiting for internet connection.
 /// - failed: An error occured. It contains AVPlayer's error if any.
-public enum AudioPlayerState {
-    case buffering
-    case playing
-    case paused
-    case stopped
-    case waitingForConnection
-    case failed(AudioPlayerError)
+@objc public enum AudioPlayerState: Int {
```

Comment:
If we do not change .failed(AudioPlayerError) in AudioPlayerState, then the whole delegate protocol will not work from obj-c.

Then we'd have to make a completely separate delegate (and state) for obj-c clients like so:
```
@objc protocol ObjcAudioPlayerDelegate {
    func audioPlayer(_ audioPlayer: AudioPlayer, didChangeStateFrom from: ObjcAudioPlayerState, to state: ObjcAudioPlayerState)
    ...
}
```
The player would then also need an objcDelegate and objcState variable as well.

Is that what you'd prefer to go with?